### PR TITLE
AnnotationUtils#getElementKindsForElementType: Ignore ElementKind.MODULE

### PR DESCRIPTION
**Pull request by smillst**

### CODE REVIEW

File: framework/tests/polyall/polyall.astub [hunk 1]
```diff
@@ -4,6 +4,8 @@ import polyall.quals.*;
 
 public final class Object {
     public Class<@H1S1 ? extends @H1S1 Object> getClass();
+    @SuppressWarnings("")
```
Comment (wmdietl) [inline framework/tests/polyall/polyall.astub]: How is this change relevant?

File: javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java [hunk 2]
```diff
@@ -447,6 +447,9 @@ public static boolean hasInheritedMeta(AnnotationMirror anno) {
             case TYPE_USE:
                 return EnumSet.noneOf(ElementKind.class);
             default:
+                if (elementType.name().contentEquals("MODULE")) {
```
Comment (wmdietl) [inline javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java]: It would be good to document this in the method javadoc.
Is this a temporary workaround or should this method never support `MODULE`?

File: javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java [hunk 3]
```diff
@@ -445,8 +445,12 @@ public static boolean hasInheritedMeta(AnnotationMirror anno) {
             case TYPE_PARAMETER:
                 return EnumSet.of(ElementKind.TYPE_PARAMETER);
             case TYPE_USE:
+            case MODULE:
```
Comment (smillst) [inline javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java]: This doesn't compile with Java 8.

File: javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java (hunk 2 above)
Comment (smillst) [reply javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java]: This temporary, which I've now noted above.

Comment (wmdietl) [reply javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java]: This is also caused by source level 8. It is a bit surprising that the source level has an impact on this, but when I change the build to not pass `source 8` I can directly refer to `ElementKind.MODULE` in the switch.
As we still want to stay on Java 8, I think this is fine, but should be documented here.

Comment (wmdietl) [reply javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java]: It's already appropriately documented.
